From f2b9cdced09ded382efad0faec7a433d38fd5e62 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Thu, 18 Sep 2014 14:14:14 -0400
Subject: [scsi] megaraid_sas: Add missing initial call to megasas_get_ld_vf_affiliation()

Message-id: <1411049668-28841-15-git-send-email-thenzl@redhat.com>
Patchwork-id: 96161
O-Subject: [RHEL7.1 e-stor PATCH 14/28] megaraid_sas: Add missing initial call to megasas_get_ld_vf_affiliation().
Bugzilla: 1088523
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

The following patch for megaraid_sas adds a missing initial call to
megasas_get_ld_vf_affiliation() at the end of megasas_probe_one().

Signed-off-by: Adam Radford <aradford@gmail.com>
Reviewed-by: Martin K. Petersen <martin.petersen@oracle.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit e06e07cc79815b32832b57e9c5b6482b021d9448)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/megaraid/megaraid_sas_base.c b/drivers/scsi/megaraid/megaraid_sas_base.c
index 8d78a8f..0e01d4a 100644
--- a/drivers/scsi/megaraid/megaraid_sas_base.c
+++ b/drivers/scsi/megaraid/megaraid_sas_base.c
@@ -4992,6 +4992,10 @@ retry_irq_register:
   goto fail_start_aen;
  }
 
+ /* Get current SR-IOV LD/VF affiliation */
+ if (instance->requestorId)
+  megasas_get_ld_vf_affiliation(instance, 1);
+
  return 0;
 
       fail_start_aen:
-- 
1.7.1